exporter/datadogexporter/logs_exporter_test.go

```diff
@@ -230,6 +230,9 @@ func TestLogsExporter(t *testing.T) {
 						Endpoint: server.URL,
 					},
 				},
+				HostMetadata: HostMetadataConfig{
+					ReporterPeriod: 30 * time.Minute,
+				},
```

Comment:
There are 8 places where you set a `HostMetadataConfig` without `Enabled: true`. My understanding is that when creating the config struct manually, the default config is not taken into account, and this will set `Enabled: false`, making the whole section have no effect. Could you confirm whether sending host metadata is actually required in those tests? If no, you can probably remove the `HostMetadata` field altogether.

Reply:
You are correct that when creating the config struct manually, the default config isn't built, and `Enabled` defaults to `false`. However, it's not the entire section that has no effect.

In the test, there's a call to `exp, err := f.CreateLogs(ctx, params, cfg)`, which creates the logs exporter via `func (f *factory) createLogsExporter`. This will call `func (f *factory) Reporter`, which instantiates the reporter with `inframetadata.NewReporter(params.Logger, pusher, pcfg.ReporterPeriod)`.

At this point, if `pcfg.ReporterPeriod` isn't initialized from the `HostMetadataConfig`, it defaults to `0`. This causes a panic in the internal reporter of `inframetadata` because we can't initialize its time ticker `time.NewTicker(period)` with a period equal to `0`.

So, even though `Enabled` is `false`, we still need to initialize `ReporterPeriod` to prevent a runtime panic. Removing the `HostMetadata` field altogether would lead to an uninitialized `ReporterPeriod`, causing the issue.

Now, you might wonder what the purpose of `Enabled` is. In fact, it's used later during the consumption of logs, as seen here: [datadogexporter/logs_exporter.go#L104](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/datadogexporter/logs_exporter.go#L104).

I hope this clarifies why we included the `HostMetadataConfig` without explicitly setting `Enabled: true`.

Reply:
I see, thank you for the detailed explanation. In other words, it seems to me that `Enabled: false` disables the calls to `Reporter.ConsumeX`, so no host data is recorded from incoming signals, but it doesn't disable the creation of a `Reporter` in the first place, or the call to `Reporter.Run`, which periodically pushes that empty data to the backend, so a non-zero `ReporterPeriod` is still needed for the corresponding `Ticker`.

Reply:
Let's go with your current solution for this PR. We can implement a more elegant solution (disabling the creation of the `Reporter` when `enabled: false`) and remove the workaround later.